packages/ag-charts-community/src/chart/errorBar.ts

```diff
@@ -0,0 +1,125 @@
+    yLowerName?: string = undefined;
+
+    @Validate(OPT_STRING)
+    yUpperKey: string = '';
+
+    @Validate(OPT_STRING)
+    yUpperName?: string = undefined;
+
+    constructor(yLowerKey: string, yUpperKey: string) {
+        this.yLowerKey = yLowerKey;
+        this.yUpperKey = yUpperKey;
+    }
+}
+
+export class ErrorBarNode extends Path {
+    public points: ErrorBarPoints = { yLowerPoint: { x: 0, y: 0 }, yUpperPoint: { x: 0, y: 0 } };
+
+    updatePath() {
+        // TODO(olegat) implement, this is just placeholder draw code
+        const { path } = this;
+        this.fill = 'black';
+        this.stroke = 'black';
+        this.strokeWidth = 1;
+        this.fillOpacity = 1;
+        this.strokeOpacity = 1;
+
+        const { yLowerPoint, yUpperPoint } = this.points;
+
+        path.clear();
+        path.moveTo(yLowerPoint.x, yLowerPoint.y);
+        path.lineTo(yUpperPoint.x, yUpperPoint.y);
+        path.closePath();
+    }
+}
+
+export class ErrorBars {
+    public groupNode?: Group = undefined;
+    private selection?: Selection<ErrorBarNode> = undefined;
+    private nodeData: (ErrorBarPoints | undefined)[] = new Array(0);
+
+    createNodeData(
+        parent: Node,
+        processedData: ProcessedData<any>,
+        xIndex?: number,
+        _yIndex?: number, // This is will be used when the scatterplot errorbars are implemented
+        xScale?: Scale<any, any, any>,
+        yScale?: Scale<any, any, any>,
+        config?: ErrorBarConfig
+    ) {
+        this.selection = undefined;
+
+        const { nodeData } = this;
+        const { yLowerKey = undefined, yUpperKey = undefined } = config ?? {};
+        if (!xScale || !yScale || !yLowerKey || !yUpperKey || xIndex === undefined || _yIndex === undefined) {
+            return;
+        }
+
+        const convert = (scale: Scale<any, any, any>, value: any) => {
+            const offset = (scale.bandwidth ?? 0) / 2;
+            return scale.convert(value) + offset;
+        };
+
+        nodeData.length = processedData.data.length;
+
+        for (let i = 0; i < processedData.data.length; i++) {
+            const { datum, values } = processedData.data[i];
+            const xDatum = values[xIndex];
+
+            if (yLowerKey in datum && yUpperKey in datum) {
+                nodeData[i] = {
+                    yLowerPoint: { x: convert(xScale, xDatum), y: convert(yScale, datum[yLowerKey]) },
+                    yUpperPoint: { x: convert(xScale, xDatum), y: convert(yScale, datum[yUpperKey]) },
+                };
+            } else {
+                nodeData[i] = undefined;
+            }
+        }
+
+        if (this.groupNode === undefined) {
+            this.groupNode = new Group({ name: `${parent.id}-series-errorBars` });
+            parent.appendChild(this.groupNode);
+        }
+
+        this.selection = Selection.select(this.groupNode, () => this.errorBarFactory());
+    }
+
+    update() {
+        if (this.selection != undefined) {
```

Comment:
This guard should be redundant if you can apply my suggestion regarding the `selection` field above 🤞

Reply:
Done.